google-cloud-core-grpc/src/test/java/com/google/cloud/grpc/BaseGrpcServiceExceptionTest.java

```diff
@@ -73,8 +74,8 @@ public void testBaseServiceException() {
     assertNull(serviceException.getDebugInfo());
 
     Exception cause = new IllegalArgumentException("bad arg");
-    GrpcApiException apiException =
-        new GrpcApiException(MESSAGE, cause, Code.INTERNAL, NOT_RETRYABLE);
+    ApiException apiException =
```

Comment:
Shouln't this be an InternalException? 
This is more of a question why do individual exceptions allow to set any different status code (it seems like there is 1:1 relationship between status code and exception, but currently it is possible to have InternalException with, for example, ABORTED status code)?

Is it intentional? If no, then I believe it would make sence to update gax (remove status code from individual exceptions parameters and hardcode the individual codes for each exception in constructor implementations).

Reply:
Yes this should be internal exception.

The reason that each exception can still accept a status code because the status code can either be a GrpcStatusCode or HttpStatusCode which can have completely different code values. Otherwise it would be hard-coding both status code into each constructor and the user will figure out which code to use. Current approach is acceptable because user will be rarely constructing exceptions by themselves so this is just the implementation details underlying gax.